### ISSUE TITLE
Render vars when parsing DbtModel

### ISSUE THREAD

**Issue by dojinkimm** [astronomer/astronomer-cosmos#495]
When parsing DbtModel if I have vars in sql file and I fail to get valid upstream_models. 
```
-- sql file
SELECT
    column_1,
	column_2,
FROM {{ ref('stg_users_'~ var('country_code')) }}
```  
```
-- model
- name: stg_users_{{ var("country_code") }}
  columns:
	...
```

For instance, if I have sql file and model file as above during `__post_init__` of `DbtModel` method, the referenced model is not added to config.upstream_models since it is not `jinja2.nodes.Const` type (If I have vars it is `Concat` type). I use vars in ref because I define single model configuration for multiple country codes.

```
-- something like this
 Concat(nodes=[Const(value='stg_users_'), Call(node=Name(name='var', ctx='load'), args=[Const(value='country_code')], kwargs=[], dyn_args=None, dyn_kwargs=None)]))
```
https://github.com/astronomer/astronomer-cosmos/blob/56d38b9a38ccc7fbe17cdcb2debb71e04f518c8b/cosmos/dbt/parser/project.py#L166-L175

So is it possible to receive vars in `DbtModel` and render ref value and add it to upstream_models? I want something like the following logic added

```
if isinstance(first_arg, jinja2.nodes.Concat):
    value = ''
    for node in base_node.args[0].nodes:
        if isinstance(node, jinja2.nodes.Const):
            value += node.value
        elif isinstance(node, jinja2.nodes.Call):
            if node.node.name == 'var':
                value += self.task_args[node.args[0].value]
```

**Comment by tatiana** [comment]
The feature totally makes sense, @dojinkimm, thanks for writing it down!
Would you like to contribute with a PR for this change?

**Comment by dojinkimm** [comment]
@tatiana Sure, I will try PR for this change

**Comment by dojinkimm** [comment]
@tatiana I recently opened [PR](https://github.com/astronomer/astronomer-cosmos/pull/502) but it is not reviewed yet. Do I need to do something more to be reviewed?

**Comment by tatiana** [comment]
Sorry for the delay, @dojinkimm , I'm looking into it now!

**Comment by tatiana** [comment]
@dojinkimm We released the fix as part of `1.1.1`:
https://github.com/astronomer/astronomer-cosmos/releases/tag/astronomer-cosmos-v1.1.1

However, you mentioned that you were on `0.6.8`. How likely are you to upgrade to the latest Cosmos? Please let us know if there are any blockers.
